Spokestack/SpeechContext.swift

```diff
@@ -18,8 +19,87 @@ import Foundation
     @objc public var isActive: Bool = false
     /// Speech detected indicator. Default to true for non-vad activation
     @objc public var isSpeech: Bool = true
-    /// `SpeechProcessor` instances that process audio frames from `AudioController`.
+    /// A set of `SpeechProcessor` instances that process audio frames from `AudioController`.
     public var stageInstances: [SpeechProcessor] = []
-    /// `SpeechEventListener`s that are sent `SpeechPipeline` events.
-    public var listeners: [SpeechEventListener] = []
+    /// A set of `SpeechEventListener`s that are sent `SpeechPipeline` events.
```

Comment:
I can’t implement Hashable in SpeechEventListener because of objc compatibility, so `Set` is unavailable for this use :( 
To save clients from themselves the observers will be an ordered set because I cannot imagine a usage where a client would deliberately register the same instance twice.